xbmc/video/windows/GUIWindowVideoNav.cpp

```diff
@@ -1045,6 +1057,21 @@ bool CGUIWindowVideoNav::OnClick(int iItem, const std::string &player)
     Refresh(true);
     return true;
   }
+  else if (StringUtils::StartsWithNoCase(item->GetPath(), "newvideoversion://"))
+  {
+    // dont allow update while scanning
+    if (CVideoLibraryQueue::GetInstance().IsScanningLibrary())
+    {
+      HELPERS::ShowOKDialogText(CVariant{257}, CVariant{14057});
+      return true;
+    }
+
+    std::string strVideoVersion;
+    CGUIDialogVideoVersion::NewVideoVersion(strVideoVersion);
```

Comment:
Why does `NewVideoVersion` not simply return the version string? Why the out param?

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#121-output-parameters

Reply:
The return value is version id.

Reply:
> The return value is version id.

 Then, you should follow the coding guidelines to return a struct or tuple.